docs/howto/data_management_framework/resources.rst

```diff
@@ -0,0 +1,136 @@
+Data Management Framework Resource Examples
+============================================
+
+The ``Resource`` class, in the ``idaes.dmf.resource`` package, is the base unit of information
+that is stored in the Data Management Framework. Below are examples of creating and using
+``Resource`` objects.
+
+.. contents::
+    :local:
+    :depth: 1
+
+Create a resource
+-----------------
+.. code-block:: python
+
+    from idaes.dmf.resource import Resource
+
+    r = Resource()
+    r.v["version_info"]["version"] = test_version
+    r.v["collaborators"] = [
+        {"name": "Clark Kent", "email": "ckent@dailyplanet.com"},
+        {"name": "Superman", "email": "sman@fortress.solitude.org"},
+    ]
+    r.v["sources"].append(
+        {
+            "isbn": "978-0201616224",
+            "source": 'Hunt, A. and Thomas, D., "The Pragmatic Programmer", '
+            "Addison-Wesley, 1999",
+            "date": "1999-01-01",
+        }
+    )
+    r.v["codes"].append(
+        {
+            "type": "function",
+            "name": "test_resource_full",
+            "desc": "The test function",
+            "location": "test_files.py",
+            "version": test_version,
+        }
+    )
+    r.v["datafiles"].append({"path": "/etc/passwd"})
+    r.v["aliases"] = ["test_resource_full"]
+    r.v["tags"] = ["test", "resource"]
+    r.data = {"arbitrary": {"values": [1, 2, 3]}}
+    return r
+
+
+Create a resource representing a CSV table
+------------------------------------------
+.. code-block:: python
+
+    from idaes.dmf.resource import Resource
+    from idaes.dmf.tables import Table
+
+    rsrc = Resource(type_=ResourceTypes.tabular)
+    rsrc.set_field("name", "table_test_resource")
+    rsrc.set_field("desc", "Example table")
+
+    t = Table()
+    # Assume existing comma-separated values file "data.csv", with a header row, e.g.
+    #     Index,Pressure,Temperature
+    #     1,256.12,38.7
+    #     2,125.61,73.8
+    t.read_csv("data.csv")
+    t.add_to_resource(rsrc)
+
+Create a resource representing a publication and multiple CSV tables
+---------------------------------------------------------------------
+In this case, the publication is the source for 10 related CSV tables.
```

Comment:
So, this is hardly important for this example, but more a discussion of concepts. The original paper was in fact 17 related tables of data, however in cleaning the data I compiled some similar tables into 1. Notably, Tables 8, 9, 12, 13, 14, 15, 16 and 17 were combined into pitzer_8.csv and Tables 10 and 11 became pitzer_10. Also note that due to this there was no pitzer_9 (and thus there were only 9 files for the example). To further complicate things, I could have done the same thing with Tables 1-7, but chose not to as they had multiple data rows per table with similar names.

So, conceptually, is is better to just to 17 separate tables as it was in the original paper (where many tables have three indexing columns and a single data column), or to do the compiled version where tables with the same indexing rows are grouped together?

Reply:
To me the most descriptive way to do this would be to represent the entire process in the DMF. But the practical answer I think is to combine as you see fit and put some of this in the description field.

Reply:
@andrewlee94 there is currently no "preprocessing script" capability for the DMF, though of course you could write one and add it as a linked resource yourself. I feel like the data should be immediately usable from whatever form they are in within the Git repo. On that note, there is certainly room to put arbitrary amounts of instructions for what each dataset means, or how it should be used, in description fields.